fog-azure 0.0.2 - including dependencies

dLobatog opened the pull request:
Not yet ready - it's missing the dependencies of fog-azure. 
This needs to be merged before https://github.com/theforeman/foreman-packaging/pull/1136

- [x] fog-azure
- [x] azure
- [x] azure-core
- [x] faraday_middleware
- [x] systemu
- [x] uuid
- [x] macaddr

domcleal commented on rubygem-fog-azure/fog-azure.spec:
```diff
@@ -0,0 +1,101 @@
+Group: Development/Languages
+License: MIT
+URL: http://github.com/fog/fog-azure
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
```
This needs to be HTTP for our builds to run.

domcleal commented on rubygem-fog-azure/fog-azure.spec:
```diff
@@ -0,0 +1,101 @@
+
+%description
+This library can be used as a module for `fog` or as standalone provider
+to use the Azure cloud services in applications..
```
stray `.` at the end

domcleal commented on rubygem-fog-azure/fog-azure.spec:
```diff
@@ -0,0 +1,101 @@
+%{?scl:EOF}
+
+# %%{?scl:scl enable %{scl} - << \EOF}
+%gem_install
```
There are two gem_installs in this section, one needs to go.

domcleal commented on rubygem-fog-azure/fog-azure.spec:
```diff
@@ -0,0 +1,101 @@
+# %%{?scl:scl enable %{scl} - << \EOF}
+%gem_install
+%{?scl:EOF}
+ compiles any C extensions and installs the gem into ./%%gem_dir
```
and missing a comment?

domcleal commented on rubygem-fog-azure/fog-azure.spec:
```diff
@@ -0,0 +1,101 @@
+
+# Run the test suite
+
+%check
```
remove this, there's no test

domcleal commented on rubygem-fog-azure/fog-azure.spec:
```diff
@@ -0,0 +1,101 @@
+%files
+%dir %{gem_instdir}
+%exclude %{gem_instdir}/.gitignore
+%license %{gem_instdir}/LICENSE.md
```
%license will fail on EL6, use %doc for now

dLobatog commented:
I've just added azure and fixed the comments - I'll make a list in the first comment with all the dependencies this needs

dLobatog commented:
Updated to include all dependencies but UUID

dLobatog commented:
Alright, all dependencies are in the PR now - waiting for Jenkins to tell me what to fix next :smile:

dLobatog commented:
Tons of wrongly prefixed ruby/rubygems packages - fixing..

domcleal commented:
Note that the next version of `azure` will drop the uuid (and therefore macaddr) dep.

I see some test failures, so guessing you're still working on this?

dLobatog commented:
Yeah, still unfinished

dLobatog commented:
Still some failures happening - but now I can reproduce them with mock `fg: no job control` - will pick it up again later.

domcleal commented:
> fg: no job control

This usually indicates you're using an unknown macro somewhere because bash is interpreting it as a job number instead of RPM replacing it. Check they're all correct and that you have any BuildRequires providing them (e.g. foreman-plugin, rubygems-devel).

dLobatog commented:
Alright :smile: last update should work just fine - I was able to build all packages locally in order (systemu, uuid, macaddr, faraday_middleware, azure, fog-azure)

domcleal commented on rubygem-macaddr/rubygem-macaddr.spec:
```diff
@@ -0,0 +1,77 @@
+
+%files
+%dir %{gem_instdir}
+%license %{gem_instdir}/LICENSE
```
%doc for EL6 compatibility

domcleal commented on rel-eng/tito.props:
```diff
@@ -566,7 +582,13 @@ whitelist = rubygem-algebrick
+  rubygem-fog-azure
+  rubygem-macaddr
   rubygem-newt
+  rubygem-openscap
```
Not needed here?

domcleal commented:
azure, azure-core seem to be missing from comps, which will stop them from being tested. I've tweaked the PR job to try and get all packages tested for your next update, as it's normally limited to five.

domcleal commented on rubygem-systemu/rubygem-systemu.spec:
```diff
@@ -0,0 +1,78 @@
+
+%files
+%dir %{gem_instdir}
+%{gem_instdir}/BSDL
```
also %doc, it's a licence

domcleal commented on rubygem-systemu/rubygem-systemu.spec:
```diff
@@ -0,0 +1,78 @@
+%{gem_instdir}/BSDL
+%doc %{gem_instdir}/LICENSE
+%{gem_libdir}
+%{gem_instdir}/samples
```
This should be in -doc, it's not required at runtime

domcleal commented on rubygem-azure-core/rubygem-azure-core.spec:
```diff
@@ -0,0 +1,89 @@
+%files
+%dir %{gem_instdir}
+%{_bindir}/pfxer
+%{gem_instdir}/.env_sample
```
move to -doc

domcleal commented on rubygem-azure-core/rubygem-azure-core.spec:
```diff
@@ -0,0 +1,89 @@
+%{!?scl:%global pkg_name %{name}}
+
+# Generated from azure-0.7.5.gem by gem2rpm -*- rpm-spec -*-
+%global gem_name azure
```
`azure-core`, this might not be building the right gem!

domcleal commented on rubygem-azure-core/rubygem-azure-core.spec:
```diff
@@ -0,0 +1,89 @@
+BuildArch: noarch
+
+%description
+Microsoft Azure Client Library for Ruby.
```
"Client Core Library"

domcleal replied:
"This project provides a Ruby package with core functionality consumed by Azure SDK gems."

domcleal commented on rubygem-azure-core/rubygem-azure-core.spec:
```diff
@@ -0,0 +1,89 @@
+Summary: Official ruby client library to consume Microsoft Azure services
+Group: Development/Languages
+License: Apache License, Version 2.0
+URL: http://github.com/azure/azure-sdk-for-ruby
```
http://github.com/Azure/azure-ruby-asm-core

domcleal commented on rubygem-azure-core/rubygem-azure-core.spec:
```diff
@@ -0,0 +1,89 @@
+Release: 1%{?dist}
+Summary: Official ruby client library to consume Microsoft Azure services
+Group: Development/Languages
+License: Apache License, Version 2.0
```
`ASL 2.0`

domcleal commented on rubygem-azure-core/rubygem-azure-core.spec:
```diff
@@ -0,0 +1,89 @@
+Name: %{?scl_prefix}rubygem-%{gem_name}
+Version: 0.7.5
+Release: 1%{?dist}
+Summary: Official ruby client library to consume Microsoft Azure services
```
"Microsoft Azure Client Core Library for Ruby SDK"

domcleal replied:
Please swap this comment with https://github.com/theforeman/foreman-packaging/pull/1140#discussion_r66590923!

domcleal commented on rubygem-azure/rubygem-azure.spec:
```diff
@@ -0,0 +1,104 @@
+Name: %{?scl_prefix}rubygem-%{gem_name}
+Version: 0.7.5
+Release: 1%{?dist}
+Summary: Official ruby client library to consume Microsoft Azure services
```
I think you've got summary/description swapped, they should be the other way around.

domcleal replied:
Ignore that, it's correct.

domcleal commented on rubygem-azure/rubygem-azure.spec:
```diff
@@ -0,0 +1,104 @@
+Release: 1%{?dist}
+Summary: Official ruby client library to consume Microsoft Azure services
+Group: Development/Languages
+License: Apache License, Version 2.0
```
ASL 2.0

domcleal commented on rubygem-azure-core/rubygem-azure-core.spec:
```diff
@@ -0,0 +1,89 @@
+%global gem_name azure
+
+Name: %{?scl_prefix}rubygem-%{gem_name}
+Version: 0.7.5
```
The .gem in the dir is version 0.1.1, or you could use 0.1.2.  0.7.5 is azure, not azure-core.

domcleal commented on rubygem-azure-core/rubygem-azure-core.spec:
```diff
@@ -0,0 +1,89 @@
+%{?scl:%scl_package rubygem-%{gem_name}}
+%{!?scl:%global pkg_name %{name}}
+
+# Generated from azure-0.7.5.gem by gem2rpm -*- rpm-spec -*-
```
it's incorrect, remove?

domcleal commented on rubygem-azure/rubygem-azure.spec:
```diff
@@ -0,0 +1,104 @@
+%files
+%dir %{gem_instdir}
+%{_bindir}/pfxer
+%{gem_instdir}/.env_sample
```
move to -doc I think

domcleal commented on rubygem-faraday_middleware/rubygem-faraday_middleware.spec:
```diff
@@ -0,0 +1,78 @@
+
+Requires: %{?scl_prefix}rubygem(faraday) < 0.10
+Requires: %{?scl_prefix}rubygem(faraday) >= 0.7.4
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
```
missing corresponding runtime Requires for ruby(release) etc.

domcleal commented on rubygem-fog-azure/rubygem-fog-azure.spec:
```diff
@@ -0,0 +1,89 @@
+URL: http://github.com/fog/fog-azure
+Source0: http://rubygems.org/gems/%{gem_name}-%{version}.gem
+Requires: %{?scl_prefix_ruby}ruby(release)
+Requires: %{?scl_prefix_ruby}rubygems-devel
```
not a runtime dependency, it's usually ruby(rubygems) and ruby

domcleal commented on rubygem-macaddr/rubygem-macaddr.spec:
```diff
@@ -0,0 +1,77 @@
+Source0: http://rubygems.org/gems/%{gem_name}-%{version}.gem
+Requires: %{?scl_prefix}rubygem(systemu) < 2.7.0
+Requires: %{?scl_prefix}rubygem(systemu) >= 2.6.2
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
```
missing corresponding runtime deps on ruby, ruby(release) and rubygems

domcleal commented on rubygem-macaddr/rubygem-macaddr.spec:
```diff
@@ -0,0 +1,77 @@
+%dir %{gem_instdir}
+%license %{gem_instdir}/LICENSE
+%{gem_libdir}
+%{gem_instdir}/rvmrc.example
```
move to -doc

domcleal commented on rubygem-systemu/rubygem-systemu.spec:
```diff
@@ -0,0 +1,78 @@
+License: Ruby
+URL: https://github.com/ahoward/systemu
+Source0: http://rubygems.org/gems/%{gem_name}-%{version}.gem
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
```
missing corresponding runtime deps on ruby(release), ruby and rubygems

domcleal commented on rubygem-uuid/rubygem-uuid.spec:
```diff
@@ -0,0 +1,86 @@
+
+Requires: %{?scl_prefix}rubygem(macaddr) >= 1.0
+Requires: %{?scl_prefix}rubygem(macaddr) < 2.0
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
```
missing corresponding runtime deps on ruby(release), ruby, rubygems

domcleal commented on rubygem-uuid/rubygem-uuid.spec:
```diff
@@ -0,0 +1,86 @@
+
+%files
+%dir %{gem_instdir}
+%{_bindir}/uuid
```
I don't think we should ship this as it'll conflict with the `uuid` package (https://apps.fedoraproject.org/packages/uuid/). Consider leaving out the binary script or renaming it.

dLobatog commented:
Updated with:

* exclusion of the uuid binary
* inclusion of ruby/rubygems runtime deps where needed
* azure-core spec actually made from azure-core (it used to be just azure, hence why it was wrong)
* updated comps to include azure and azure-core
* moved various random files to -doc packages

dLobatog commented:
:+1: tests passed!

domcleal commented on rubygem-azure-core/rubygem-azure-core.spec:
```diff
@@ -0,0 +1,87 @@
+Group: Development/Languages
+License: ASL 2.0
+URL: http://github.com/Azure/azure-ruby-asm-core
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
```
I don't think the PR test is checking this gem (I'll double check why), but this will need to be http:// for it to work.

domcleal replied:
It was testing the gem but not picking up errors from `tito release`, Jenkins job now fixed!

domcleal commented on rubygem-faraday_middleware/rubygem-faraday_middleware.spec:
```diff
@@ -0,0 +1,81 @@
+%{?scl:EOF}
+
+%{?scl:scl enable %{scl} - << \EOF}
+%gem_install -n %{SOURCE0}
```
Remove `-n %{SOURCE0}` here, as you've built a new .gem in the prep/build stages and using SOURCE0 skips that (useful) work by going straight back to the .gem you unpacked.

domcleal commented on rubygem-fog-azure/rubygem-fog-azure.spec:
```diff
@@ -0,0 +1,89 @@
+gem build %{gem_name}.gemspec
+%{?scl:EOF}
+%{?scl:scl enable %{scl} - << \EOF}
+%gem_install -n %{SOURCE0}
```
Remove `-n %{SOURCE0}` here.

domcleal commented on rubygem-systemu/rubygem-systemu.spec:
```diff
@@ -0,0 +1,81 @@
+%{?scl:EOF}
+
+%{?scl:scl enable %{scl} - << \EOF}
+%gem_install -n %{SOURCE0}
```
Remove `-n %{SOURCE0}` here.

domcleal commented on rubygem-uuid/rubygem-uuid.spec:
```diff
@@ -0,0 +1,88 @@
+%{?scl:EOF}
+
+%{?scl:scl enable %{scl} - << \EOF}
+%gem_install -n %{SOURCE0}
```
Remove `-n %{SOURCE0}` here.

domcleal commented:
All the gems apart from fog-azure are also missing their `Provides:... rubygem(%{gem_name})...` line, which means the dependencies between them probably won't work.

domcleal commented on rubygem-azure/rubygem-azure.spec:
```diff
@@ -0,0 +1,105 @@
+Requires: %{?scl_prefix}rubygem(faraday_middleware) < 1.0
+Requires: %{?scl_prefix_ruby}rubygem(json) >= 1.8
+Requires: %{?scl_prefix_ruby}rubygem(json) < 2.0
+Requires: %{?scl_prefix}rubygem(mime-types) >= 1.0
```
This doesn't resolve, it should be `scl_prefix_ror` I think.

dLobatog commented:
Updated so that `mime-types` comes from `scl_prefix_ror`

domcleal commented on comps/comps-foreman-plugins-fedora24.xml:
```diff
@@ -71,11 +73,16 @@
       <packagereq type="default">rubygem-graphite-api</packagereq>
       <packagereq type="default">rubygem-hammer_cli_foreman_ssh</packagereq>
       <packagereq type="default">rubygem-logify</packagereq>
+      <packagereq type="default">rubygem-macaddr</packagereq>
```
This can be removed, it's in Fedora (https://apps.fedoraproject.org/packages/rubygem-macaddr).

domcleal commented on comps/comps-foreman-plugins-fedora24.xml:
```diff
@@ -71,11 +73,16 @@
+      <packagereq type="default">rubygem-macaddr</packagereq>
       <packagereq type="default">rubygem-newt</packagereq>
       <packagereq type="default">rubygem-satyr</packagereq>
+      <packagereq type="default">rubygem-systemu</packagereq>
```
ditto (https://apps.fedoraproject.org/packages/rubygem-systemu)

domcleal commented on comps/comps-foreman-plugins-fedora24.xml:
```diff
@@ -71,11 +73,16 @@
       <packagereq type="default">rubygem-newt</packagereq>
       <packagereq type="default">rubygem-satyr</packagereq>
+      <packagereq type="default">rubygem-systemu</packagereq>
+      <packagereq type="default">rubygem-uuid</packagereq>
```
ditto (https://apps.fedoraproject.org/packages/rubygem-uuid)

domcleal commented on rel-eng/tito.props:
```diff
@@ -578,7 +594,12 @@ whitelist = rubygem-algebrick
+  rubygem-macaddr
   rubygem-newt
+  rubygem-systemu
+  rubygem-uuid
```
also remove them from here

domcleal commented:
Thanks, it installs and appears to work now.

dLobatog commented:
Removed macaddr/systemu/uuid from Fedora

domcleal commented:
Thanks @dLobatog, merged and beginning to build.
